### ISSUE TITLE
Index tensor must have the same number of dimensions as src tensor

### ISSUE THREAD

**Issue by toolgood** [dotnet/TorchSharp#600]
Code migration from Python
``` python
    def _one_hot(self, labels, classes, value=1):
        one_hot = torch.zeros(labels.size(0), classes)
        #labels and value_added  size must match
        labels = labels.view(labels.size(0), -1)
        value_added = torch.Tensor(labels.size(0), 1).fill_(value)

        value_added = value_added.to(labels.device)
        one_hot = one_hot.to(labels.device)
        one_hot.scatter_add_(1, labels, value_added)
        return one_hot
```
CSharp code:
``` C#
      private Tensor _one_hot(Tensor labels, int classes=8, double value = 1)
        {
            var one_hot = torch.zeros(labels.size(0), classes); ;
            labels = labels.view(labels.size(0), -1);
            var value_added = torch.tensor(labels.size(0), 1).fill_(value).to(torch.float32);
            value_added = value_added.to(labels.device);
            one_hot = one_hot.to(labels.device);
            one_hot.scatter_add_(1, labels, value_added); //error
            return one_hot;
        }
```

labels.shape
{long[1]}
    [0]: 256

Error message:
System.Runtime.InteropServices.ExternalException:“Index tensor must have the same number of dimensions as src tensor
Exception raised from scatter_shape_check at ..\..\aten\src\ATen/native/ScatterGatherChecks.h:107 (most recent call first):
00007FFC3292A29200007FFC3292A230 c10.dll!c10::Error::Error [<unknown file> @ <unknown line number>]
00007FFC32929CAE00007FFC32929C60 c10.dll!c10::detail::torchCheckFail [<unknown file> @ <unknown line number>]
00007FFC093ABC3900007FFC093AAF80 torch_cpu.dll!at::native::quantized_index [<unknown file> @ <unknown line number>]
00007FFC0939004F00007FFC0938B440 torch_cpu.dll!at::native::_bincount_cpu [<unknown file> @ <unknown line number>]
00007FFC093A9F8F00007FFC093A9EF0 torch_cpu.dll!at::meta::structured_scatter_add::meta [<unknown file> @ <unknown line number>]
00007FFC09AE36DD00007FFC09AC8130 torch_cpu.dll!at::cpu::zero_ [<unknown file> @ <unknown line number>]
00007FFC095CCDAB00007FFC0954BF30 torch_cpu.dll!at::TensorMaker::make_tensor [<unknown file> @ <unknown line number>]
00007FFC0990FDC600007FFC0990FD40 torch_cpu.dll!at::_ops::scatter_add_::redispatch [<unknown file> @ <unknown line number>]
00007FFC0B19D91800007FFC0AB97570 torch_cpu.dll!at::redispatch::zeros_outf [<unknown file> @ <unknown line number>]
00007FFC095CCDAB00007FFC0954BF30 torch_cpu.dll!at::TensorMaker::make_tensor [<unknown file> @ <unknown line number>]
00007FFC0990FDC600007FFC0990FD40 torch_cpu.dll!at::_ops::scatter_add_::redispatch [<unknown file> @ <unknown line number>]
00007FFC0AD3787F00007FFC0AB97570 torch_cpu.dll!at::redispatch::zeros_outf [<unknown file> @ <unknown line number>]
00007FFC098CBF3600007FFC098CBDF0 torch_cpu.dll!at::_ops::scatter_add_::call [<unknown file> @ <unknown line number>]
00007FFC30DA4BE000007FFC30DA4B60 LibTorchSharp.DLL!THSTensor_scatter_add_ [<unknown file> @ <unknown line number>]
00007FFBD1F0B0A3 <unknown symbol address> !<unknown symbol> [<unknown file> @ <unknown line number>]

**Comment by NiklasGustafsson** [comment]
@toolgood -- thanks for the issue report. Could you add the code that calls _one_hot()? From the printout, I'm guessing it's a tensor with a single element '256', is that right?

**Comment by toolgood** [comment]
have solved this bug

```
        private static Tensor _one_hot(Tensor labels, int classes, double value = 1)
        {
            var one_hot = torch.zeros(labels.size(0), classes); ;
            labels = labels.view(labels.size(0), -1);
            var value_added = torch.full(new long[] { labels.size(0), 1 }, value, ScalarType.Float32);

            value_added = value_added.to(labels.device);
            one_hot = one_hot.to(labels.device);
            one_hot.scatter_add_(1, labels, value_added);
            return one_hot;
        }
```
